obj.rb

```diff
@@ -0,0 +1,34 @@
+require 'gosu'
+require 'chipmunk'
+require_relative 'chip-gosu-functions'
+
+class Obj
+  attr_reader :shapes, :body
+  def initialize(window, g, source)
+    @image = Gosu::Image.new window, source
+    @g = g
+    @shapes = []
+  end
+
+  def warp(vect)
+    @shapes[0].body.p = vect
+  end
+
+  def add_to_space(space)
+    space.add_body @body
+    @shapes.each do |shape|
+      space.add_shape shape
+    end
+  end
+
+  def do_gravity(power)
+    @shapes.each do |shape|
+      shape.body.apply_force((CP::Vec2.new(0.0, 1.0) * (power)), CP::Vec2.new(0.0, 0.0)) if @g
+    end
+  end
+  def draw
+    fx = @sizex * 1.0 / @image.width
+    fy = @sizey * 1.0 / @image.height
+    @image.draw_rot(@shapes[0].body.p.x, @shapes[0].body.p.y, 0, @shapes[0].body.a.radians_to_gosu, 0, 0, fx, fy)
```

Comment:
This is still not great, but it's better. I would do it like this:

```
position = [@shapes[0].body.p.x, @shapes[0].body.p.y]
rotation = ...
function = [fx, fy]
@image.draw_rot *(position + rotation + function)
```

I don't know if I split them nicely, but you get the general idea. You should do this everywhere you have such long argument lists.